Tests/SonarScanner.MSBuild.Tasks.UnitTests/GetAnalyzerSettingsTests.cs

```diff
@@ -384,8 +361,54 @@ public void MergeRulesets_OriginalRulesetSpecified_AbsolutePath_SecondGeneratedR
             ExecuteAndCheckSuccess(testSubject);
 
             // Assert
-            CheckMergedRulesetFile(testSubject, "e:\\sub1\\originalRuleset.txt",
-                "firstGeneratedRuleset.txt");
+            CheckMergedRulesetFile(testSubject, "e:\\sub1\\originalRuleset.txt");
+        }
+
+        [TestMethod]
+        // Regression test for #581: Sonar issues are reported as external issues
+        // https://github.com/SonarSource/sonar-scanner-msbuild/issues/581
+        public void MergeRuleset_CheckQPSettingsWin()
+        {
+            // Arrange
+            // Off in QP, on locally -> off
+            var qpRulesetPath = CreateRuleset("qpRuleset", @"<?xml version='1.0' encoding='utf-8'?>
+<RuleSet xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' Name='x' Description='x' ToolsVersion='14.0'>
+  <Rules AnalyzerId='analyzer1' RuleNamespace='ns1'>
+    <Rule Id='SharedRuleOffInQP' Action='None' />
+    <Rule Id='SharedRuleOffInLocal' Action='Warning' />
+    <Rule Id='QPOnlyRule' Action='Info' />
+  </Rules>
+</RuleSet>");
+
+            var localRulesetPath = CreateRuleset("localRuleset", @"<?xml version='1.0' encoding='utf-8'?>
+<RuleSet xmlns:xsd='http://www.w3.org/2001/XMLSchema' xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' Name='x' Description='x' ToolsVersion='14.0'>
+  <Rules AnalyzerId='analyzer1' RuleNamespace='ns1'>
+    <Rule Id='SharedRuleOffInQP' Action='Error' />
+    <Rule Id='SharedRuleOffInLocal' Action='None' />
+    <Rule Id='LocalOnlyRule' Action='Error' />
+  </Rules>
+</RuleSet>");
+
+            var config = CreateMergingAnalysisConfig("xxx", qpRulesetPath);
+
+            var testSubject = CreateConfiguredTestSubject(config, "xxx", TestContext);
+            testSubject.OriginalRulesetFilePath = localRulesetPath;
+            testSubject.ProjectSpecificConfigDirectory = testSubject.AnalysisConfigDir;
+
+            // Act
+            ExecuteAndCheckSuccess(testSubject);
+
+            // Assert
+            CheckMergedRulesetFile(testSubject, localRulesetPath);
+
+            // Check individual rule severities are set correctly
+            var finalPath = testSubject.RuleSetFilePath;
+            var actualRuleset = MSCA.RuleSet.LoadEffectiveRuleSetFromFile(finalPath);
```

Comment:
Using the _Microsoft.CodeAnalysis.RuleSet_ class to load and validate the generated ruleset.
The method will throw if the file is invalid. It also calculates the effective severities to use for the rules, so we can use it to check that the behaviour of the merged ruleset.